docs/architecture/workflow-update.md

```diff
@@ -203,51 +211,58 @@ stateDiagram-v2
     notFound: (nil, NotFoundError)
     unavailable: (nil, UnavailableError)
     
-    [*] --> completed: onResponseMsg
-    [*] --> failed: onResponseMsg
-    [*] --> rejected: onRejectionMsg
-    [*] --> rejected: RejectUnprocessed
+    [*] --> completed: onResponseMsg()
+    [*] --> failed: onResponseMsg()
+    [*] --> rejected: onRejectionMsg()
+    [*] --> rejected: RejectUnprocessed()
     [*] --> notFound: abort(reason=WorkflowCompleted)
     [*] --> unavailable: abort(reason=RegistryCleared)
 ```
 
-### Internal Timeout
-Also, caller can specify a timeout it's willing to wait before specified stage is reached. If timeout
-is expired and Update didn't reach the stage, `context deadline exceeded` error will be returned to the caller.
+### Timeouts
+The API caller can specify the time it is willing to wait before the specified stage is reached.
+If the timeout expires and the Update has not reach the desired stage yet, a 
+`context deadline exceeded` error will be returned to the caller.
 
-But if caller doesn't specify timeout, server will enforce it to `LongPollExpirationInterval`
-(default is 20 seconds). When this timeout is expired, server won't return `context deadline exceeded` error,
-but, instead, will return an empty response with actual reached stage. SDK has special logic to handle
-this response:
-- If reached stage is `ADMITTED`, means that Update wasn't persisted yet, and might be lost, SDK
-retries `UpdateWorkflowExecution` API call,
-- If reached stage is `ACCEPTED`, then there is no reason to retry original call and SDK starts to poll
-Update results using `PollWorkflowExecutionUpdate` API.
+If the API caller does not specify a timeout, the server will enforce a `LongPollExpirationInterval`
+(default is 20 seconds). When this timeout expires, the server won't return a 
+`context deadline exceeded` error, but instead will return an empty response with the reached stage.
 
-> The same empty response with actual reached stage is returned when the Update Registry is cleared,
+The client needs to handle the result as follows:
+- If reached stage is `ADMITTED`, it means that the Update was not persisted yet and might be lost -
+the client should retry the `UpdateWorkflowExecution` API call.
+- If reached stage is `ACCEPTED`, it means there is no reason to retry the call - the client should
+start polling for the Update result using the `PollWorkflowExecutionUpdate` API.
+
+[TODO]
+> #### NOTE
+> The same empty response with the reached stage is returned when the Update Registry is cleared,
 > but Update reached `ACCEPTED` stage. Instead of retrying `UpdateWorkflowExecution` API, SDK starts to poll.
 > If reached stage is `ADMITTED` only, server returns retryable `Unavailable` error, which shouldn't
 > reach SDK and retried internally on server, but if it does, SDK behavior is the same as empty
 > response with `ADMITTED` stage: retry `UpdateWorkflowExecution` API call. 
```

Comment:
If registry is cleared behavior is almost the same, but instead of returning the empty response it return the Unavailable error.